perceval/backends/mozilla/mozillaclub.py

```diff
@@ -0,0 +1,397 @@
+        origin = url
+        self.url = url
+        super().__init__(origin, tag=tag, cache=cache)
+        self.client = MozillaClubClient(url)
+        self.__users = {}  # internal users cache
+
+    @metadata
+    def fetch(self):
+        """Fetch events from the MozillaClub url.
+
+        The method retrieves, from a MozillaClub url, the
+        events. The data is a Google spreadsheet retrieved using
+        the feed API REST.
+
+        :returns: a generator of events
+        """
+
+        logger.info("Looking for events at url '%s'", self.url)
+
+        nevents = 0  # number of events processed
+        nevents_wrong = 0  # number of events with wrong data
+
+        self._purge_cache_queue()
+
+        raw_cells = self.client.get_cells()
+        self._push_cache_queue(raw_cells)
+        sheet_json = json.loads(raw_cells)
+        self._flush_cache_queue()
+
+        parser = MozillaClubParser()
+
+        for event in parser.parse(sheet_json):
+            yield event
+            nevents += 1
+
+        logger.info("Total number of events: %i", nevents)
+
+    @metadata
+    def fetch_from_cache(self):
+        """Fetch the events from the cache.
+
+        :returns: a generator of events
+
+        :raises CacheError: raised when an error occurs accessing the
+            cache
+        """
+
+        logger.info("Retrieving cached events: '%s'", self.url)
+
+        if not self.cache:
+            raise CacheError(cause="cache instance was not provided")
+
+        cache_items = next(self.cache.retrieve())
+        sheet_json = json.loads(cache_items)
+
+        parser = MozillaClubParser()
+
+        nevents = 0
+
+        for event in parser.parse(sheet_json):
+            yield event
+            nevents += 1
+
+        logger.info("Total number of events from cache: %i", nevents)
+
+    @classmethod
+    def has_caching(cls):
+        """Returns whether it supports caching items on the fetch process.
+
+        :returns: this backend supports items cache
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend supports items resuming
+        """
+        return False
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from an event item."""
+
+        try:
+            a = item.keys()
+        except AttributeError:
+            print ("->"+item+"<-")
+            raise
+        return str(item['Date of Event']+"_"+item['Club Name'])
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a item.
+
+        This backend only generates one type of item which is
+        'event'.
+        """
+        return 'event'
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a MozillaClub item.
+
+        The timestamp is extracted from 'end' field.
```

Comment:
Fixed. Here it is a problem: this date is not the real update date for the item. Probably we must use the unixtime for the collect timestamp. What do you think @sduenas?